highlighters/main/main-highlighter.zsh

```diff
@@ -442,16 +509,20 @@ _zsh_highlight_highlighter_main_paint()
     # Parse the sudo command line
     if (( ! in_redirection )); then
       if [[ $this_word == *':sudo_opt:'* ]]; then
-        case "$arg" in
+        if [[ -n $flags_with_argument ]] &&
+           [[ $arg == '-'[$flags_with_argument] ]]; then
           # Flag that requires an argument
-          '-'[Cgprtu]) this_word=${this_word//:start:/};
-                       next_word=':sudo_arg:';;
+          this_word=${this_word//:start:/}
```

Comment:
Any reason to not use `this_word=':sudo_arg:'` (here and below)?

Reply:
Where's below?  Did you mean `this_word=':sudo_opt:'`?